tempodb/encoding/common/types.go

```diff
@@ -31,6 +31,7 @@ type Appender interface {
 	Complete()
 	Records() []*Record
 	Length() int
```

Comment:
We should be close to being able to remove the `Length()` function on this interface.  Double check and see how hard it would be? 

If nothing else the AppendBlock doesn't need the Length() method anymore and it can be removed.

Reply:
Yes it is mostly unused now.  The remaining use is for compactor max records per block.  As discussed, we should probably move that to a file size-based approach as well.